build error under conda on Windows 7

I am trying to use pip to install pyhacrf under conda on Windows 7 and it is giving me the following error. Any help with a work around would be greatly apprecieated.

```
C:\Users\User\AppData\Local\Continuum\Anaconda>pip install pyhacrf
Collecting pyhacrf
  Using cached pyhacrf-0.0.12.tar.gz
Requirement already satisfied (use --upgrade to upgrade): numpy>=1.9 in c:\users\User\appdata\local\continuum\anaconda\lib\site-p
ackages (from pyhacrf)
Requirement already satisfied (use --upgrade to upgrade): PyLBFGS>=0.1.3 in c:\users\User\appdata\local\continuum\anaconda\lib\si
te-packages (from pyhacrf)
Building wheels for collected packages: pyhacrf
  Running setup.py bdist_wheel for pyhacrf
  Complete output from command C:\Users\User\AppData\Local\Continuum\Anaconda\python.exe -c "import setuptools;__file__='c:\\user
s\\User\\appdata\\local\\temp\\pip-build-rojt7c\\pyhacrf\\setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __f
ile__, 'exec'))" bdist_wheel -d c:\users\User\appdata\local\temp\tmpqnoh1dpip-wheel-:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-2.7
  creating build\lib.win-amd64-2.7\pyhacrf
  copying pyhacrf\feature_extraction.py -> build\lib.win-amd64-2.7\pyhacrf
  copying pyhacrf\pyhacrf.py -> build\lib.win-amd64-2.7\pyhacrf
  copying pyhacrf\state_machine.py -> build\lib.win-amd64-2.7\pyhacrf
  copying pyhacrf\__init__.py -> build\lib.win-amd64-2.7\pyhacrf
  running build_ext
  Looking for python27.dll
  building 'pyhacrf.algorithms' extension
  C compiler: gcc -m64 -g -DNDEBUG -DMS_WIN64 -O2 -Wall -Wstrict-prototypes

  creating build\temp.win-amd64-2.7
  creating build\temp.win-amd64-2.7\Release
  creating build\temp.win-amd64-2.7\Release\pyhacrf
  compile options: '-D__MSVCRT_VERSION__=0x0900 -IC:\\Users\\User\\AppData\\Local\\Continuum\\Anaconda\\lib\\site-packages\\numpy
\\core\\include -IC:\Users\User\AppData\Local\Continuum\Anaconda\lib\site-packages\numpy\core\include -IC:\Users\User\AppData\
Local\Continuum\Anaconda\include -IC:\Users\User\AppData\Local\Continuum\Anaconda\PC -c'
  gcc -m64 -g -DNDEBUG -DMS_WIN64 -O2 -Wall -Wstrict-prototypes -D__MSVCRT_VERSION__=0x0900 -IC:\\Users\\User\\AppData\\Local\\Co
ntinuum\\Anaconda\\lib\\site-packages\\numpy\\core\\include -IC:\Users\User\AppData\Local\Continuum\Anaconda\lib\site-packages\nu
mpy\core\include -IC:\Users\User\AppData\Local\Continuum\Anaconda\include -IC:\Users\User\AppData\Local\Continuum\Anaconda\PC
-c pyhacrf/algorithms.c -o build\temp.win-amd64-2.7\Release\pyhacrf\algorithms.o
  Found executable C:\Users\User\AppData\Local\Continuum\Anaconda\Scripts\gcc.bat
  gcc -m64 -g -shared build\temp.win-amd64-2.7\Release\pyhacrf\algorithms.o -LC:\\Users\\User\\AppData\\Local\\Continuum\\Anacond
a\\lib\\site-packages\\numpy\\core\\lib -LC:\Users\User\AppData\Local\Continuum\Anaconda\libs -LC:\Users\User\AppData\Local\Co
ntinuum\Anaconda\PCbuild\amd64 -lnpymath -lpython27 -lmsvcr90 -o build\lib.win-amd64-2.7\pyhacrf\algorithms.pyd
  Warning: .drectve `/manifestdependency:"type='win32' name='Microsoft.VC90.CRT' version='9.0.21022.8' processorArchitecture='amd64'
 publicKeyToken='1fc8b3b9a1e18e3b'" /DEFAULTLIB:"python27.lib" /DEFAULTLIB:"MSVCRT" /DEFAULTLIB:"OLDNAMES" ' unrecognized
  Warning: .drectve `/manifestdependency:"type='win32' name='Microsoft.VC90.CRT' version='9.0.21022.8' processorArchitecture='amd64'
 publicKeyToken='1fc8b3b9a1e18e3b'" /DEFAULTLIB:"python27.lib" /DEFAULTLIB:"MSVCRT" /DEFAULTLIB:"OLDNAMES" ' unrecognized
  C:\\Users\\User\\AppData\\Local\\Continuum\\Anaconda\\lib\\site-packages\\numpy\\core\\lib/npymath.lib(build/temp.win-amd64-2.7
/build/src.win-amd64-2.7/numpy/core/src/npymath/npy_math.obj):(.text+0x2e3): undefined reference to `__imp_modff'
  collect2.exe: error: ld returned 1 exit status
  error: Command "gcc -m64 -g -shared build\temp.win-amd64-2.7\Release\pyhacrf\algorithms.o -LC:\\Users\\User\\AppData\\Local\\Co
ntinuum\\Anaconda\\lib\\site-packages\\numpy\\core\\lib -LC:\Users\User\AppData\Local\Continuum\Anaconda\libs -LC:\Users\User\
AppData\Local\Continuum\Anaconda\PCbuild\amd64 -lnpymath -lpython27 -lmsvcr90 -o build\lib.win-amd64-2.7\pyhacrf\algorithms.pyd" fai
led with exit status 1

  ----------------------------------------
  Failed building wheel for pyhacrf
```

Reading this it looks like the problem is in `/npy_math.obj`

```
  C:\\Users\\User\\AppData\\Local\\Continuum\\Anaconda\\lib\\site-packages\\numpy\\core\\lib/npymath.lib(build/temp.win-amd64-2.7
/build/src.win-amd64-2.7/numpy/core/src/npymath/npy_math.obj):(.text+0x2e3): undefined reference to `__imp_modff'
```

Can you look for help from conda on this?

--- a/pyhacrf/algorithms.pyx
+++ b/pyhacrf/algorithms.pyx
@@ -6,9 +6,9 @@ from numpy import ndarray
 from numpy cimport ndarray
 from numpy.math cimport logaddexp, INFINITY as inf
 cdef extern from "math.h" nogil :
-    double exp(double x)
+    np.float64_t exp(np.float64_t x)
 
-cpdef dict forward(np.ndarray[long, ndim=2] lattice, np.ndarray[double, ndim=3] x_dot_parameters, long S):
+cpdef dict forward(np.ndarray[np.int64_t, ndim=2] lattice, np.ndarray[np.float64_t, ndim=3] x_dot_parameters, long S):
     """ Helper to calculate the forward weights.  """
     cdef dict alpha = {}
 
@@ -17,7 +17,7 @@ cpdef dict forward(np.ndarray[long, ndim=2] lattice, np.ndarray[double, ndim=3] 
     cdef unsigned int I, J, s
 
     cdef unsigned int old_i0, old_j0, old_s0 
-    cdef double edge_potential
+    cdef np.float64_t edge_potential
 
     old_i0, old_j0, old_s0 = -1, -1, -1
 
@@ -35,9 +35,9 @@ cpdef dict forward(np.ndarray[long, ndim=2] lattice, np.ndarray[double, ndim=3] 
             old_i0, old_j0, old_s0 = i0, j0, s0
 
         edge_potential = (x_dot_parameters[i1, j1, edge_parameter_index]
-                          + <double> alpha[(i0, j0, s0)])
+                          + <np.float64_t> alpha[(i0, j0, s0)])
         alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] = edge_potential
-        alpha[(i1, j1, s1)] = logaddexp(<double> alpha.get((i1, j1, s1), -inf), 
+        alpha[(i1, j1, s1)] = logaddexp(<np.float64_t> alpha.get((i1, j1, s1), -inf),
                                         edge_potential)
 
     I = x_dot_parameters.shape[0] - 1
@@ -47,16 +47,16 @@ cpdef dict forward(np.ndarray[long, ndim=2] lattice, np.ndarray[double, ndim=3] 
         if I == J == 0:
             alpha[(I, J, s)] = x_dot_parameters[I, J, s]
         else:
-            alpha[(I, J, s)] = <double> alpha.get((I, J, s), -inf) + x_dot_parameters[I, J, s]
+            alpha[(I, J, s)] = <np.float64_t> alpha.get((I, J, s), -inf) + x_dot_parameters[I, J, s]
 
     return alpha
 
-cpdef double[:, :, ::1] forward_predict(long[:, ::1] lattice, 
-                                      double[:, :, ::1] x_dot_parameters, 
+cpdef np.float64_t[:, :, ::1] forward_predict(np.int64_t[:, ::1] lattice,
+                                      np.float64_t[:, :, ::1] x_dot_parameters,
                                       long S) :
     """ Helper to calculate the forward weights for prediction.  """
 
-    cdef double[:, :, ::1] alpha = x_dot_parameters.copy()
+    cdef np.float64_t[:, :, ::1] alpha = x_dot_parameters.copy()
     alpha[:] = -inf
 
     cdef unsigned int r 
@@ -64,7 +64,7 @@ cpdef double[:, :, ::1] forward_predict(long[:, ::1] lattice,
 
     cdef int old_s0 = -1
 
-    cdef double edge_potential, source_node_potential
+    cdef np.float64_t edge_potential, source_node_potential
 
     for r in range(lattice.shape[0]):
         i0, j0, s0 = lattice[r, 0], lattice[r, 1], lattice[r, 2]
@@ -97,12 +97,12 @@ cpdef double[:, :, ::1] forward_predict(long[:, ::1] lattice,
     return alpha
 
 
-cpdef double[:, :, ::1] forward_max_predict(long[:, ::1] lattice,
-                                            double[:, :, ::1] x_dot_parameters,
+cpdef np.float64_t[:, :, ::1] forward_max_predict(np.int64_t[:, ::1] lattice,
+                                            np.float64_t[:, :, ::1] x_dot_parameters,
                                             long S) :
     """ Helper to calculate the forward max-sum weights for prediction.  """
 
-    cdef double[:, :, ::1] alpha = x_dot_parameters.copy()
+    cdef np.float64_t[:, :, ::1] alpha = x_dot_parameters.copy()
     alpha[:] = -inf
 
     cdef unsigned int r
@@ -110,7 +110,7 @@ cpdef double[:, :, ::1] forward_max_predict(long[:, ::1] lattice,
 
     cdef int old_s0 = -1
 
-    cdef double edge_potential, source_node_potential
+    cdef np.float64_t edge_potential, source_node_potential
 
     for r in range(lattice.shape[0]):
         i0, j0, s0 = lattice[r, 0], lattice[r, 1], lattice[r, 2]
@@ -143,8 +143,8 @@ cpdef double[:, :, ::1] forward_max_predict(long[:, ::1] lattice,
     return alpha
 
 
-cpdef dict backward(ndarray[long, ndim=2] lattice,
-                    ndarray[double, ndim=3] x_dot_parameters,
+cpdef dict backward(ndarray[np.int64_t, ndim=2] lattice,
+                    ndarray[np.float64_t, ndim=3] x_dot_parameters,
                     long I, long J, long S):
     """ Helper to calculate the backward weights.  """
     cdef dict beta = {}
@@ -153,7 +153,7 @@ cpdef dict backward(ndarray[long, ndim=2] lattice,
     cdef unsigned int s
     cdef unsigned int i0, j0, s0, i1, j1, s1, edge_parameter_index
 
-    cdef double edge_potential
+    cdef np.float64_t edge_potential
 
     for s in range(S):
         beta[(I-1, J-1, s)] = 0.0
@@ -163,9 +163,9 @@ cpdef dict backward(ndarray[long, ndim=2] lattice,
         i1, j1, s1 = lattice[r, 3], lattice[r, 4], lattice[r, 5]
         edge_parameter_index = lattice[r, 6]
 
-        edge_potential = <double> beta[(i1, j1, s1)] + x_dot_parameters[i1, j1, s1]
+        edge_potential = <np.float64_t> beta[(i1, j1, s1)] + x_dot_parameters[i1, j1, s1]
         beta[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] = edge_potential
-        beta[(i0, j0, s0)] = logaddexp(<double> beta.get((i0, j0, s0), -inf),
+        beta[(i0, j0, s0)] = logaddexp(<np.float64_t> beta.get((i0, j0, s0), -inf),
                                        (edge_potential 
                                         + x_dot_parameters[i1, 
                                                            j1, 
@@ -175,31 +175,31 @@ cpdef dict backward(ndarray[long, ndim=2] lattice,
 
 def gradient(dict alpha,
              dict beta,
-             ndarray[double, ndim=2] parameters,
-             ndarray[long] states_to_classes,
-             ndarray[double, ndim=3] x,
+             ndarray[np.float64_t, ndim=2] parameters,
+             ndarray[np.int64_t] states_to_classes,
+             ndarray[np.float64_t, ndim=3] x,
              long y,
              long I, long J, long K):
     """ Helper to calculate the marginals and from that the gradient given the forward and backward weights. """
     cdef unsigned int n_classes = max(states_to_classes) + 1
-    cdef ndarray[double] class_Z = np.zeros((n_classes,))
-    cdef double Z = -inf
-    cdef double weight
+    cdef ndarray[np.float64_t] class_Z = np.zeros((n_classes,))
+    cdef np.float64_t Z = -inf
+    cdef np.float64_t weight
     cdef unsigned int k
 
     for state, clas in enumerate(states_to_classes):
-        weight = <double> alpha[(I - 1, J - 1, state)]
+        weight = <np.float64_t> alpha[(I - 1, J - 1, state)]
         class_Z[clas] = weight
         Z = logaddexp(Z, weight)
 
-    cdef ndarray[double, ndim=2] derivative = np.full_like(parameters, 0.0)
+    cdef ndarray[np.float64_t, ndim=2] derivative = np.full_like(parameters, 0.0)
     cdef unsigned int i0, j0, s0, i1, j1, s1, edge_parameter_index
-    cdef double alphabeta
+    cdef np.float64_t alphabeta
 
     for node in alpha.viewkeys() | beta.viewkeys():
         if len(node) == 3:
             i0, j0, s0 = node
-            alphabeta = <double>alpha[(i0, j0, s0)] + <double>beta[(i0, j0, s0)]
+            alphabeta = <np.float64_t>alpha[(i0, j0, s0)] + <np.float64_t>beta[(i0, j0, s0)]
 
             for k in range(K):
                 if states_to_classes[s0] == y:
@@ -209,8 +209,8 @@ def gradient(dict alpha,
 
         else:
             i0, j0, s0, i1, j1, s1, edge_parameter_index = node
-            alphabeta = <double>alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] \
-                        + <double>beta[(i0, j0, s0, i1, j1, s1, edge_parameter_index)]
+            alphabeta = <np.float64_t>alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] \
+                        + <np.float64_t>beta[(i0, j0, s0, i1, j1, s1, edge_parameter_index)]
 
             for k in xrange(K):
                 if states_to_classes[s1] == y:
@@ -223,10 +223,10 @@ def gradient(dict alpha,
 
 def gradient_sparse(dict alpha,
                     dict beta,
-                    ndarray[double, ndim=2] parameters,
-                    ndarray[long] states_to_classes,
-                    ndarray[long, ndim=3] x_index,
-                    ndarray[double, ndim=3] x_value,
+                    ndarray[np.float64_t, ndim=2] parameters,
+                    ndarray[np.int64_t] states_to_classes,
+                    ndarray[np.int64_t, ndim=3] x_index,
+                    ndarray[np.float64_t, ndim=3] x_value,
                     long y,
                     long I, long J, long K):
     """
@@ -234,26 +234,26 @@ def gradient_sparse(dict alpha,
     sparse input features.
     """
     cdef unsigned int n_classes = max(states_to_classes) + 1
-    cdef ndarray[double] class_Z = np.zeros((n_classes,))
-    cdef double Z = -inf
-    cdef double weight
+    cdef ndarray[np.float64_t] class_Z = np.zeros((n_classes,))
+    cdef np.float64_t Z = -inf
+    cdef np.float64_t weight
     cdef unsigned int C = K
     cdef unsigned int c
     cdef int k
 
     for state, clas in enumerate(states_to_classes):
-        weight = <double> alpha[(I - 1, J - 1, state)]
+        weight = <np.float64_t> alpha[(I - 1, J - 1, state)]
         class_Z[clas] = weight
         Z = logaddexp(Z, weight)
 
-    cdef ndarray[double, ndim=2] derivative = np.full_like(parameters, 0.0)
+    cdef ndarray[np.float64_t, ndim=2] derivative = np.full_like(parameters, 0.0)
     cdef unsigned int i0, j0, s0, i1, j1, s1, edge_parameter_index
-    cdef double alphabeta
+    cdef np.float64_t alphabeta
 
     for node in alpha.viewkeys() | beta.viewkeys():
         if len(node) == 3:
             i0, j0, s0 = node
-            alphabeta = <double>alpha[(i0, j0, s0)] + <double>beta[(i0, j0, s0)]
+            alphabeta = <np.float64_t>alpha[(i0, j0, s0)] + <np.float64_t>beta[(i0, j0, s0)]
 
             for c in range(C):
                 k = x_index[i0, j0, c]
@@ -266,8 +266,8 @@ def gradient_sparse(dict alpha,
 
         else:
             i0, j0, s0, i1, j1, s1, edge_parameter_index = node
-            alphabeta = <double>alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] \
-                                + <double>beta[(i0, j0, s0, i1, j1, s1, edge_parameter_index)]
+            alphabeta = <np.float64_t>alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] \
+                                + <np.float64_t>beta[(i0, j0, s0, i1, j1, s1, edge_parameter_index)]
 
             for c in range(C):
                 k = x_index[i1, j1, c]
@@ -281,9 +281,9 @@ def gradient_sparse(dict alpha,
     return (class_Z[y]) - (Z), derivative
 
 
-def populate_sparse_features(ndarray[double, ndim=3] x,
-                             ndarray[long, ndim=3] index_array,
-                             ndarray[double, ndim=3] value_array,
+def populate_sparse_features(ndarray[np.float64_t, ndim=3] x,
+                             ndarray[np.int64_t, ndim=3] index_array,
+                             ndarray[np.float64_t, ndim=3] value_array,
                              long I, long J, long K):
     """ Helper to fill in sparse feature arrays. """
     cdef unsigned int i, j, c, k
@@ -296,10 +296,10 @@ def populate_sparse_features(ndarray[double, ndim=3] x,
                     index_array[i, j, c] = k
                     c += 1
 
-def sparse_multiply(ndarray[double, ndim=3] answer,
-                    ndarray[long, ndim=3] index_array,
-                    ndarray[double, ndim=3] value_array,
-                    ndarray[double, ndim=2] dense_array,
+def sparse_multiply(ndarray[np.float64_t, ndim=3] answer,
+                    ndarray[np.int64_t, ndim=3] index_array,
+                    ndarray[np.float64_t, ndim=3] value_array,
+                    ndarray[np.float64_t, ndim=2] dense_array,
                     long I, long J, long K, long C, long S):
     """ Multiply a sparse three dimensional numpy array (using our own scheme) with a two dimensional array. """
     cdef unsigned int i, j, s, c

--- a/pyhacrf/feature_extraction.py
+++ b/pyhacrf/feature_extraction.py
@@ -134,7 +134,7 @@ class PairFeatureExtractor(object):
         K = (len(self._binary_features) 
              + sum(num_feats for _, num_feats in self._sparse_features))
 
-        feature_array = np.zeros((array1.size, array2.size, K))
+        feature_array = np.zeros((array1.size, array2.size, K), dtype='float64')
 
         for k, feature_function in enumerate(self._binary_features):
             feature_array[..., k] = feature_function(array1, array2)

--- a/pyhacrf/pyhacrf.py
+++ b/pyhacrf/pyhacrf.py
@@ -104,7 +104,7 @@ class Hacrf(object):
                 ll += dll
                 gradient += dgradient
 
-            parameters_without_bias = np.array(parameters)  # exclude the bias parameters from being regularized
+            parameters_without_bias = np.array(parameters, dtype='float64')  # exclude the bias parameters from being regularized
             parameters_without_bias[0] = 0
             ll -= self.l2_regularization * np.dot(parameters_without_bias.T, parameters_without_bias)
             gradient = gradient.flatten() - 2.0 * self.l2_regularization * parameters_without_bias

--- a/pyhacrf/pyhacrf.py
+++ b/pyhacrf/pyhacrf.py
@@ -259,7 +259,7 @@ class _Model(object):
         beta = self._backward(x_dot_parameters)
         classes_to_ints = {k: i for i, k in enumerate(set(self.states_to_classes.values()))}
         states_to_classes = np.array([classes_to_ints[self.states_to_classes[state]]
-                                      for state in range(max(self.states_to_classes.keys()) + 1)])
+                                      for state in range(max(self.states_to_classes.keys()) + 1)], dtype='int64')
         if not isinstance(self.sparse_x, str):
             ll, deriv = gradient_sparse(alpha, beta, parameters, states_to_classes,
                                         self.sparse_x[0], self.sparse_x[1], classes_to_ints[self.y],

--- a/pyhacrf/state_machine.py
+++ b/pyhacrf/state_machine.py
@@ -84,7 +84,7 @@ class GeneralStateMachine(object):
 
         # Squash list
         lattice = [edge for edge in reversed_list if len(edge) > 3]
-        return np.array(lattice)
+        return np.array(lattice, dtype='int64')
 
 
 class DefaultStateMachine(object):
@@ -158,7 +158,7 @@ class DefaultStateMachine(object):
         lattice += _grow_independent_lattice(self._transitions, 
                                              self.n_states, (I, J), 
                                              unvisited_nodes)
-        lattice = np.array(sorted(lattice), dtype=int)
+        lattice = np.array(sorted(lattice), dtype='int64')
         return lattice
 
     def build_lattice(self, x):

